source/common/quic/envoy_quic_session_cache.cc

```diff
@@ -0,0 +1,168 @@
+#include "common/quic/envoy_quic_session_cache.h"
+
+namespace Envoy {
+namespace Quic {
+namespace {
+
+constexpr size_t kMaxSessionCacheEntries = 1024;
```

Comment:
Envoy doesn't use kCamelCaseConstantName convention for const. MAX_SESSION_CACHE_SIZE instead?
Do this needs to be configuration? @alyssawilk

Reply:
I went with `MaxSessionCacheEntries` which is preferred by the [Envoy style guide](https://github.com/envoyproxy/envoy/blob/main/STYLE.md) because it doesn't conflict with macros

Reply:
we'll want config eventually but that can be a TODO (either for David, against QUIC upstream, I vote for the former =P) with a comment that the hard-coded value is arbitrary

Also @RyanTheOptimist look, another bounded cache!  now that I look they're everywhere =P

Reply:
I can file an issue to make this configurable once this lands. Though I don't think it's a priority

Reply:
I'm thinking more that having configurable 0-rtt config and integration tests is worthwhile for envoy mobile :-)

Reply:
What's a configurable 0-RTT config?
This PR adds an integration test.